### PR TITLE
Accessibility improvements for links and buttons

### LINKED ISSUE #6404: review link/button text and expand or add labels where necessary (WCAG 2.4.4, 3.3.2)
## Description

[Accessibility Lab recommendation](https://docs.google.com/document/d/1jZlKco2YgOrrI4PmTkG_Q8lKuV5VxCds/edit?disco=AAAAIplLpuE):

> The purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context, except where the purpose of the link would be ambiguous to users in general.

## User Research Evidence

[WCAG 2.4.4](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) per #5972.

## Implementation

Suggested as part of accessibility review of #6315:
* When a user navigates using a screen-reader, do existing text/labels give sufficient context for where links will take them and what button will do ("Delete user `journalist`")? or must they infer from surrounding elements ("Delete user")?
* "Read more" links should have `aria-label`s that expand to "Read more about [...]"

### PR DESCRIPTION
**Pull request by nathandyer**
## Status

Ready for review

## Description of Changes

Fixes #6404

Changes proposed in this pull request:

* This PR adds and expands labels to improve accessibility in areas where existing descriptions were either missing, or lacking context for visually impaired users to be able to understand where they are in the system and where they will be navigating next.
* ~This PR also includes updated translation files, as new translations will be needed.~

## Testing

How should the reviewer test this PR?

* Build the dev environment using the usual steps, then verify everything is working as expected.
* Optionally, enable a screen reader to verify these changes and see (or rather, hear) the differences first-hand.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

This will require new translations.

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container*
 \* There are three failures with make test, but it appears to be unrelated to changes in this branch, and I still see those when I run `make test` for the `develop` branch as well. 653 tests pass.

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [x] These changes do not require documentation

### CODE REVIEW

File: securedrop/source_templates/login.html
Comment (cfm) [inline securedrop/source_templates/login.html]: Nit here too, per <https://docs.securedrop.org/en/stable/source.html#making-your-first-submission>:

```suggestion
    <button type="submit" aria-label="{{ gettext('Continue to Submission Page') }}">
```

File: securedrop/source_templates/logout.html
Comment (cfm) [inline securedrop/source_templates/logout.html]: Could this be simplified to either of:

```suggestion
  <a href="{{ url_for('main.login') }}" class="btn" aria-label="{{ gettext('Log In Again') }}">
```

"Again" makes sense here!  But at that point a source already knows what's involved.  Or:

```suggestion
  <a href="{{ url_for('main.login') }}" class="btn" aria-label="{{ gettext('Log Back In' }}">
```

File: securedrop/source_templates/index.html
Comment (cfm) [inline securedrop/source_templates/index.html]: How would you feel about:

```suggestion
            aria-label="{{ gettext('Log In Using Your Codename') }}">
```

"Previous" makes sense from our system-level perspective.  But for a source I imagine it as "my" rather than "a previous" codename.

File: securedrop/source_templates/generate.html
Comment (cfm) [inline securedrop/source_templates/generate.html]: Nit:  It looks like the most-canonical term we have for this page is "submission page", singular, per <https://docs.securedrop.org/en/stable/source.html#making-your-first-submission>.  (Though I would bikeshed in favor of "submission form"!)

```suggestion
  <button type="submit" aria-label="{{ gettext('Continue to Submission Page') }}">
```

File: securedrop/source_templates/lookup.html
Comment (cfm) [inline securedrop/source_templates/lookup.html]: Nit: We tend not to use the definite article before "SecureDrop", in which case:

```suggestion
      <button type="submit" aria-label="{{ gettext('Submit Uploaded Content to SecureDrop') }}">
```

I wonder about emulating [`base.html`'s logic around the `organization_name`](https://github.com/freedomofpress/securedrop/blob/f7ca802b6842e8af812f71cc9dca2800e0089183/securedrop/source_templates/base.html#L8-L12).  But I think this is fine, as would be the even simpler:

```suggestion
      <button type="submit" aria-label="{{ gettext('Submit Uploaded Content') }}">
```

File: securedrop/source_templates/base.html
Comment (cfm) [inline securedrop/source_templates/base.html]: I worry that "main view" is a technical concept that may not help users understand where this link will take them.  How would you feel about making it explicit, as part of the Jinja conditional?

* if `is_user_logged_in`: "Return to Submission Page" (as elsewhere)
* else: "Return to Home Page" (another page for which I don't think we have a canonical term)

File: securedrop/source_templates/utils.html
Comment (cfm) [inline securedrop/source_templates/utils.html]: Hmm: it might not _meaningfully_ be, though, for a user who's navigating exclusively by screen-reader.  What about:

```suggestion
    aria-label="{{ gettext('Reveal Codename') }}" aria-controls="codename">
```

In fact, I wonder if "reveal" is the ability- and interface-agnostic term for what this widget does here.  Just thinking out loud.  :-)

File: securedrop/journalist_templates/base.html
Comment (cfm) [inline securedrop/journalist_templates/base.html]: I have the same reservation about the "main view" concept here as in the Source Interface.  How about this, per <https://docs.securedrop.org/en/stable/journalist.html#interacting-with-sources>:

```suggestion
        <a aria-label="{{ gettext('Return to All Sources') }}" href="{{ url_for('main.index') }}" class="no-bottom-border"><img src="{{ g.logo }}" class="logo small"
```

File: securedrop/journalist_templates/index.html
Comment (cfm) [inline securedrop/journalist_templates/index.html:25]: These make such a difference; thank you!

File: securedrop/journalist_templates/col.html
Comment (cfm) [inline securedrop/journalist_templates/col.html:30]: These make such a difference; thank you!

**Review by cfm — CHANGES_REQUESTED** [review]
Thanks for your work here, @nathandyer!  I apologize for taking so long to get around to reviewing this pull request.  Your checklist looks good:

* [x] Build the dev environment using the usual steps, then verify everything is working as expected.
* [x] Optionally, enable a screen reader to verify these changes and see (or rather, hear) the differences first-hand.

I've left some nit-level suggestions inline.  Please feel free to take what's useful and ignore me where I'm just bikeshedding.  :-)

In addition, rather than resolving the conflicts between `develop` and your 53b52b13758dcfc44a712128b5c73925a044b4c1, I'd suggest you just amend that commit to skip `make translate` altogether.  Despite our [documentation](https://developers.securedrop.org/en/latest/i18n.html#update-strings-to-be-translated), we're not truly in the habit of updating source strings as we go, and pending #6232 there's no real advantage to doing so.  So it will be simplest now (for review) and in the next release cycle (to see what's changed) if we defer `make translate` to the localization freeze.

File: securedrop/source_templates/index.html
Comment (nathandyer) [reply securedrop/source_templates/index.html]: I agree, this makes more sense when thinking from the perspective of a Source

File: securedrop/source_templates/utils.html
Comment (nathandyer) [reply securedrop/source_templates/utils.html]: That's a great distinction, and I think your instinct of using "reveal" is ideal here.

File: securedrop/source_templates/logout.html
Comment (nathandyer) [reply securedrop/source_templates/logout.html]: I think the suggestion of "Log Back In" is clearer here, so I propose we go with that.

File: securedrop/journalist_templates/index.html
Comment (nathandyer) [reply securedrop/journalist_templates/index.html:25]: My pleasure!

File: securedrop/journalist_templates/col.html
Comment (nathandyer) [reply securedrop/journalist_templates/col.html:30]: Happy to work on it, hopefully folks will find these changes helpful and meaningful!

File: securedrop/source_templates/base.html
Comment (nathandyer) [reply securedrop/source_templates/base.html]: I think that makes a ton of sense; implemented with https://github.com/freedomofpress/securedrop/pull/6536/commits/bac9dc2e4a78e1f48f7155c189ef43b5b6b60850

**Comment by nathandyer** [comment]
@cfm Thanks for the thorough review! I especially appreciate the explanation about when to run `make translate` - I'm in the habit of regenerating translation files after any string changes in other projects, so I carried that over here. Your explanation makes sense, and I'll hold off on doing those manually from now on (so that it can be done once at the time of the localization freeze).

I believe I've made the requested changes, and have rebased to remove the previous changes to the translation files. Please let me know if there's anything else I need to do before this is ready to merge. Thanks!

File: securedrop/source_templates/logout.html
Comment (cfm) [inline securedrop/source_templates/logout.html]: Let's see if this is sufficient to unbreak the failing tests!

```suggestion
  <a href="{{ url_for('main.login') }}" class="btn" aria-label="{{ gettext('Log Back In') }}">
```

**Review by cfm — CHANGES_REQUESTED** [review]
Looks and sounds great, @nathandyer!  I've tacked on 58f03ff8cb165a334b9ca3435b9636aed18b67dd to fix tests failing due to a missing `)` in a `gettext()` call.  Please rebase this in however you see fit, and I'll approve and merge today.

Other notes for your consideration and the record:
1. I've found two extra places where links have associated actions that are indicated by icons and could benefit from explicit labeling:
    1. https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/col.html#L67-L69
    2. https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/_source_row.html#L42-L50
    If feel like tacking on commits for these, feel free, and I'll review promptly!  Otherwise, lest the perfect be the enemy of the very good work already here, let me know, and I'll just open a follow-up ticket.  :-)
2. I thought I'd found a few other unlabeled spots in the Journalist Interface, but these turn out to all be side effects of how we template ([1](https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/_confirmation_modal.html), [2](https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/_sources_confirmation_final_modal.html), [3](https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/_sources_confirmation_modal.html)) and [localize](https://github.com/freedomofpress/securedrop/blob/58f03ff8cb165a334b9ca3435b9636aed18b67dd/securedrop/journalist_templates/js-strings.html) the Journalist Interface's progressive enhancement.  I think that's out of the scope of the drop-in improvements you've made here, so I *will* open a follow-up ticket about this more-involved step.

**Comment by cfm** [comment]
@nathandyer, if you happen to get around to these changes tomorrow, I can review this again.  Otherwise I'll stay tuned in the new year.  :-)

**Comment by nathandyer** [comment]
@cfm I believe the one-liner commit I added should address the needs in `_source_row.html`. If I need to make any changes there, just let me know (I made some assumptions about ngettext, but I believe they're reasonable assumptions). 

As we briefly discussed in a separate thread, the improvements you mentioned for `col.html` are worthwhile, but given the slightly different scope, I think it would make sense to do so outside of this PR. If you'd like to do so here, though, we certainly can.

File: securedrop/journalist_templates/_source_row.html
Comment (cfm) [inline securedrop/journalist_templates/_source_row.html:48]: Nit: extra whitespace:

```suggestion
    <span class="visually-hidden">
```

Comment (cfm) [inline securedrop/journalist_templates/_source_row.html]: Nit: extra whitespace:

```suggestion
    <span class="visually-hidden">
```

**Review by cfm — CHANGES_REQUESTED** [review]
Looks and sounds great, @nathandyer!  Left one nit re: extra whitespace in 01f43956433db14069f5b4d657c791f9b4c3cac1, but as soon as you fix that up I'll approve and merge.  :-)

**Comment by cfm** [comment]
Let's call this very belated spring cleaning....

**Review by cfm — DISMISSED** [review]

**Review by cfm — APPROVED** [review]

**Pull request merged**
